lua/orgmode/init.lua

```diff
@@ -47,17 +47,35 @@ function Org:reload(file)
 end
 
 function Org:setup_autocmds()
-  vim.cmd([[augroup orgmode_nvim]])
-  vim.cmd([[autocmd!]])
-  vim.cmd(
-    [[autocmd BufWritePost *.org,*.org_archive call luaeval('require("orgmode").reload(_A)', expand('<afile>:p'))]]
-  )
-  vim.cmd(
-    [[autocmd BufReadPost,BufWritePost *.org,*.org_archive lua require('orgmode.org.diagnostics').print_error_state()]]
-  )
-  vim.cmd([[autocmd FileType org call luaeval('require("orgmode").reload(_A)', expand('<afile>:p'))]])
-  vim.cmd([[autocmd CursorHold,CursorHoldI *.org,*.org_archive lua require('orgmode.org.diagnostics').report()]])
-  vim.cmd([[augroup END]])
+  vim.api.nvim_create_augroup('orgmode_nvim', { clear = true })
```

Comment:
Put this into variable and use the variable in the `nvim_create_autocmd` instead of string

Reply:
@ranjithshegde I see you marked this as resolved, but it's not actually changed. This is what I suggest:
```suggestion
local org_augroup = vim.api.nvim_create_augroup('orgmode_nvim', { clear = true })
```

and below in autocmds set `group = org_augroup`. Example:

```lua
  vim.api.nvim_create_autocmd('BufWritePost', {
    pattern = { '*.org', '*.org_archive' },
    group = org_augroup,
    callback = function()
      require('orgmode').reload(vim.fn.expand('<afile>:p'))
    end,
  })
```